node/client/src/lib.rs

```diff
@@ -418,66 +418,69 @@ impl sc_client_api::BlockBackend<Block> for Client {
 	}
 }
 
-impl sc_client_api::StorageProvider<Block, crate::FullBackend> for Client {
+impl sc_client_api::StorageProvider<Block, crate::FullBackend> for Client
+where
+	Block: BlockT,
+{
 	fn storage(
 		&self,
-		id: &BlockId<Block>,
+		hash: &<Block as BlockT>::Hash,
```

Comment:
```suggestion
		hash: &Block::Hash,
```

You don't need to fully qualify the trait.

Reply:
This won't compile:
```error[E0223]: ambiguous associated type
   --> node/client/src/lib.rs:427:10
    |
427 |         hash: &Block::Hash,
    |                ^^^^^^^^^^^ help: use fully-qualified syntax: `<sp_runtime::generic::Block<sp_runtime::generic::Header<u32, BlakeTwo256>, OpaqueExtrinsic> as Trait>::Hash`
```

Reply:
Ahh yeah, I thought `Block` is a generic type here. However, it is the actual type. So, this doesn't work. Sorry!